src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -110,26 +135,20 @@ public void Start(Func<Task> elapsed)
                     CancellationToken.None,
                     TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
                     TaskScheduler.Default);
-
-            CancelAndDispose(Interlocked.Exchange(ref this.tokenSource, newTokenSource));
         }
 
         /// <summary>
         /// Cancels the current task.
         /// </summary>
         public void Cancel()
         {
-            CancelAndDispose(Interlocked.Exchange(ref this.tokenSource, null));
+            DelayedWork currentWork = this.latestWork;
+            if (currentWork != null)
+            {
+                currentWork.IsCancellationRequested = true;
+            }
         }
 
-        /// <summary>
-        /// Releases unmanaged and - optionally - managed resources.
-        /// </summary>
-        public void Dispose()
```

Comment:
@macrogreg I see that the `TaskTimer.Dispose` method is now `[Obsolete]` but the `TelemetryBuffer`, `MetricManager` and other consumers no longer call it. It also doesn't look like they call `TaskTimer.Cancel` which is supposed to replace it. This is a significant change in the implementation of the `TaskTimer` consumers that may leave background tasks running, indefinitely, if the background task restarts the timer.